### ISSUE TITLE
UE5 support

### ISSUE THREAD

**Issue by dvagala** [ydrive/EasySynth#45]
Any plans to support the UE5?

I would love to help with this, but I'm just starting with the UE altogether.

**Comment by fpanjevic** [comment]
There's active development in a branch named `ue5`. It should work on UE5 with limited features (optical flow still has issues).

**Comment by dvagala** [comment]
@fpanjevic perfect! I've added now the plugin to the project, it detects it correctly, but I don't see the EasySynth button in the toolbar. Please do you know how to access the plugin in UE5?

![Screenshot 2022-11-13 at 00 24 25](https://user-images.githubusercontent.com/37017489/201499851-5e30d308-073a-4867-91ac-91d5796cf165.png)

![Screenshot 2022-11-13 at 00 24 06](https://user-images.githubusercontent.com/37017489/201499846-302134c1-57a9-48a4-8c87-f2a0257bf9a5.png)

UE Version: 5.0.3-20979098+++UE5+Release-5.0

**Comment by NikolaJov96** [comment]
Hi @dvagala, it seems that UE5 does not allow plugin buttons anymore. You can access the plugin through the `Window` menu:

![image](https://user-images.githubusercontent.com/8226562/201500169-d6d11a0f-d4fe-4718-bde8-bb6fc57170a1.png)

**Comment by NikolaJov96** [comment]
@dvagala sorry, I now see that you also could not find it in the `Window` tab. I am unsure what could be the reason, have you restarted the editor after enabling the plugin?
I can see that you are using a Mac machine, we were not able to test the plugin on a Mac, but there should be no reason for it not to work.

**Comment by dvagala** [comment]
@NikolaJov96 yes yes, I tried to restart and still not seeing the EasySynth in Window tab.

But I was now looking into `EasySynth.uplugin`'s WhitelistPlatforms and there is no Mac support. I tried to add a "Mac" there and then when starting the UE it says:

```
The following modules are missing or built with a different engine version:

EasySynth

Would you like to rebuild them now?
```

If I click `no`, UE won't load, i I click `yes` it shows error:
```
Error

MyProject3 could not be compiled. Try rebuilding from source manually.
```

**Comment by NikolaJov96** [comment]
Great catch, will add it to the supported platforms when you successfully get it working.

Not sure why the build at startup fails, but you can try building the plugin separately, and hopefully it will recognize it on startup. Have you installed the engine through the launcher or built it from the source?

If you built the engine from the source, try building the plugin with the engine by cloning the `ue5` branch to the `<ue5_installation_dir>/Engine/Plugins/` and starting the build.

If that does not work, you should be able to run the project build from the terminal like it's explained here
https://docs.unrealengine.com/4.26/en-US/SharingAndReleasing/Deployment/BuildOperations/

The command should be run from the ` /Engine/Build/BatchFiles/` directory and should look like this:
```
./RunUAT BuildCookRun -project=<full path to .uproject file> -build
```

If the build is successful, you should be able to open the project without the need to rebuild it.

**Comment by dvagala** [comment]
@NikolaJov96 I installed the engine through the launcher. So I guess that the steps you described wouldn't work right?

I'll try tomorrow to build the engine through the source code and do what you suggested. I'll let you know

**Comment by NikolaJov96** [comment]
@dvagala sorry for not being clear, I don't know if the `RunUAT` script exists in the `/Engine/Build/BatchFiles/` folder if the engine is installed through the launcher, but if it does, the build should work.

**Comment by dvagala** [comment]
@NikolaJov96 no worries. Yes the `RunUAT` script is there, I've tried to run it now, and sadly the build returns an error:

`ERROR: Platform Mac is not a valid platform to build. Check that the SDK is installed properly.`

A quick check on Google, and it seems to point out that there is some problem with my Xcode compatibility, but I'm not sure. I think now I can't afford to mess around with this because I'm using the Xcode daily for work.


Here's the full output:
```
 ./RunUAT.sh BuildCookRun -project=/Users/dvagala/Documents/Unreal\ Projects/MyProject3/MyProject3.uproject -build                                                          (base)

Running AutomationTool...

Running bundled mono, version: Mono JIT compiler version 5.16.0.220 (2018-06/bb3ae37d71a Fri Nov 16 17:12:11 EST 2018)
Setting up bundled DotNet SDK
Start UAT Interactively: ./AutomationTool BuildCookRun -project=/Users/dvagala/Documents/Unreal Projects/MyProject3/MyProject3.uproject -build
Parsing command line: BuildCookRun -project="/Users/dvagala/Documents/Unreal Projects/MyProject3/MyProject3.uproject" -build
Using xbuild. Install Mono 5.0 or greater for faster builds!
Log location: /Users/dvagala/Library/Logs/Unreal Engine/LocalBuildLogs/Log.txt
Setting up ProjectParams for /Users/dvagala/Documents/Unreal Projects/MyProject3/MyProject3.uproject
/Users/dvagala/Documents/Unreal Projects/MyProject3/Plugins/EasySynth/EasySynth.uplugin: warning: Unknown platform Win32 while parsing allow list for module descriptor EasySynth
MyProject3.uproject requires a temporary target.cs to be generated (DatasmithImporter plugin is enabled)
********** BUILD COMMAND STARTED **********
Building UnrealEditor as x86_64 (arm64 not currently supported)
Running: /Users/Shared/Epic Games/UE_5.0/Engine/Binaries/DotNET/UnrealBuildTool/UnrealBuildTool UnrealEditor Mac Development -Project="/Users/dvagala/Documents/Unreal Projects/MyProject3/MyProject3.uproject"  "/Users/dvagala/Documents/Unreal Projects/MyProject3/MyProject3.uproject" -NoUBTMakefiles  -architecture=x86_64 -Manifest="/Users/dvagala/Documents/Unreal Projects/MyProject3/Intermediate/Build/Manifest.xml" -NoHotReload -log="/Users/dvagala/Library/Logs/Unreal Engine/LocalBuildLogs/UBT-UnrealEditor-Mac-Development.txt"
Log file: /Users/dvagala/Library/Logs/Unreal Engine/LocalBuildLogs/UBT-UnrealEditor-Mac-Development.txt
ERROR: Platform Mac is not a valid platform to build. Check that the SDK is installed properly.
Took 2.4522199999999996s to run UnrealBuildTool, ExitCode=6
UnrealBuildTool failed. See log for more details. (/Users/dvagala/Library/Logs/Unreal Engine/LocalBuildLogs/UBT-UnrealEditor-Mac-Development.txt)
AutomationTool executed for 0h 0m 5s
AutomationTool exiting with ExitCode=6 (6)
RunUAT ERROR: AutomationTool was unable to run successfully. Exited with code: 6
```

**Comment by fpanjevic** [comment]
UE 5.1 support is now available in the official release 🎉

**Comment by tejasr2002** [comment]
Hi, you mentioned that optical flow has some issues on UE5. Is that still the case? Would really appreciate a confirmation.
